### ISSUE TITLE
adblocker doesn't block trafficjunky ads

### ISSUE THREAD

**Issue by leem32** [cliqz-oss/adblocker#683]
I need to remove adult content from screenshots, but the puppeteer extra adblock plugin which uses this library fails to block trafficjunky ads found on sites such as pornhub.com and redtube.com.

Traffic junky ads are inside an iframe, so that's probably the reason.

Are there any workarounds for this?

```javascript
const puppeteer = require('puppeteer-extra');
const AdblockerPlugin = require('puppeteer-extra-plugin-adblocker');

const adblocker = AdblockerPlugin({
  blockTrackers: true // default: false
});
puppeteer.use(adblocker);
```

Debug output:

```javascript

DEBUG=puppeteer-extra,puppeteer-extra-plugin:* node screenshot.js https://www.redtube.com/
puppeteer-extra-plugin:base:adblocker Initialized. +0ms
puppeteer-extra-plugin:adblocker Initialized { blockTrackers: true, useCache: true, cacheDir: undefined } +0ms
puppeteer-extra plugin registered adblocker +0ms
puppeteer-extra no dependencies are missing +2ms
puppeteer-extra orderPlugins:before [ 'adblocker' ] +0ms
puppeteer-extra orderPlugins:after [ 'adblocker' ] +1ms
puppeteer-extra-plugin:adblocker beforeLaunch +0ms
puppeteer-extra-plugin:adblocker getBlocker { hasBlocker: false } +0ms
puppeteer-extra-plugin:adblocker load from cache /tmp/puppeteer-extra-plugin-adblocker-2.11.3-engine.bin +0ms

```

https://github.com/berstend/puppeteer-extra/issues/176

**Comment by remusao** [comment]
Hey @leem32,

Thanks for opening the issue. I will have a look ASAP and try to reproduce locally. I will update the progress on this issue,

Best,

**Comment by remusao** [comment]
What I could see so far is that ads are blocked but the emplacement of their iframe is still visible like on the screenshot below. Is this what you observe as well on your side? Strangely is seems that these are actually injected by puppeteer/Chrome itself.

![ph-ads-blocked](https://user-images.githubusercontent.com/1299873/77911685-82e39a00-7291-11ea-885a-541fcc5eca2c.png)

Here from the network monitor we can see that the network requests are blocked as they should:

![ph-ads-network](https://user-images.githubusercontent.com/1299873/77911681-8119d680-7291-11ea-998f-76c6e80a1063.png)

Can you maybe confirm that you have the same thing on your side?

Best,

**Comment by remusao** [comment]
@leem32 I have investigated and fixed a few issues with `PuppeteerBlocker` thanks to your report. I also released a new version with these changes: `1.13.0`, could you maybe give it a shot and tell me if it solved your issue? Here is an highlight from the changelog:

> PuppeteerBlocker is now more powerful and will be able to better block ads on most websites. Firstly, a bug was fixed which prevented injection of cosmetics in the main frame of pages. Secondly, PuppeteerBlocker will now monitor the DOM for changes to make sure that ads which load later are still "handled" (if you know what I mean). Lastly, PuppeteerBlocker is now able to look for advertisement iframes and remove them from the DOM completely, no more blank spaces left unattended...

Thanks again!

**Comment by leem32** [comment]
@remusao I've tested it multiple times and it's now working great.

Thanks :)
